### ISSUE TITLE
Some violations not reported (like SpacingAroundOperators)

### ISSUE THREAD

**Issue by jarekratajski** [detekt/detekt#3537]
UPDATE:
No bug. Works as designed. plugins where needed.

I have created a sample project which demonstrates the problem:
https://github.com/jarekratajski/whodetectsdetekt
## Expected Behavior
I expect that such code:
```
fun main() {
    kotlin.io.println("runs")
    val x = NUM1
    if (x<NUM2) {
        println("unexpected")
    }
    if (x== NUM2) {
        println("unexpected")
    }
}
```
would cause at lease few violations but none are reported `./gradlew detekt`

## Observed Behavior
nothing is reported when gradle task is executed

## Steps to Reproduce
run `./gradlew detekt` on an example project

## Context
Problem found in a bigger project, definitely some rules work, but others do not.
Problem happens also with previous versions.

## Your Environment
check example project
same behaviour for various versions of java8 and java11 (hotspot)

**Comment by chao2zhang** [comment]
Hello @jarekratajski, `ForbiddenMethodCall` requires type resolution, you can refer to `https://detekt.github.io/detekt/type-resolution.html` on learning more. In real projects, most of the intention can be achieved with [ForbiddenImport](https://detekt.github.io/detekt/style.html#forbiddenimport) which does not require type resolution.

`SpacingAroundOperators` is a ktlint (detekt formatting) rule, you can refer to [Adding more rule sets](https://detekt.github.io/detekt/index.html#adding-more-rule-sets) on actually including ktlint (detekt formatting) checks.

**Comment by jarekratajski** [comment]
as for second ) Yes - just found - adding more rule sets. So prabably not a bug - checking that.
The question - in a real project I would like to prevent `println` - is there any other rule checking for that.

**Comment by jarekratajski** [comment]
I confirm there is no bug - all wortks. Which I updated in example project.
Thank you for quick response.

Small note - maybe it should be written somewhere on top of detekt.yml description that plugins are needed. 
I will think about it and maybe propose a documentation fix.

**Comment by chao2zhang** [comment]
One improvement idea is that we do not warn users if there are type-resolving rules that are active but the detekt is run without type-resolution / the rules are not in the plugin classpath.

**Comment by schalkms** [comment]
@chao2zhang good idea. Do you care to submit a new issue for that?

**Comment by chao2zhang** [comment]
Sounds good.
